pkg/client/restclient/config.go

```diff
@@ -118,6 +118,16 @@ type Config struct {
 	// Version string
 }
 
+// ImpersonationConfig has all the available impersonation options
+type ImpersonationConfig struct {
+	// UserName matches user.Info.GetName()
```

Comment:
This is not really descriptive. I have no clue what `user.Info.GetName()` is in this context.